src/libraries/Microsoft.PowerFx.Core/Public/Config/ComposedReadOnlySymbolTable.cs

```diff
@@ -61,29 +61,35 @@ public override FormulaType GetTypeFromSlot(ISymbolSlot slot)
         private TexlFunctionSet _nameResolverFunctions = null;
         private VersionHash _cachedVersionHash = VersionHash.New();
 
+        // To keep _cachedVersionHash and _nameResolverFunctions in sync.
+        private readonly object _lock = new object();
+
         // Expose the list to aide in intellisense suggestions. 
         // Multiple readers ok. But not writing while we read.
         TexlFunctionSet INameResolver.Functions
         {
             get
             {
-                var current = this.VersionHash;
-                if (current != _cachedVersionHash)
+                lock (_lock)
```

Comment:
@jas-valgotar  - fyi. 
This lock doesn't prevent mutations. but it should ensure the fields are updated consistently and avoid the null refs.